app/models/region.rb

```diff
@@ -1,5 +1,5 @@
 class Region < ApplicationRecord
-  has_many :subprojects
+  has_many :subprojects, dependent: :restrict_with_error
```

Comment:
@CulmoneY let me know if we want to add model-level tests for this or if the controller-based integration tests are enough